src/proof/bitvector_proof.cpp

```diff
@@ -394,7 +210,7 @@ void LFSCBitVectorProof::printBitOf(Expr term, std::ostream& os, const ProofLetM
   os << " " << bit << ")";
 }
 
-void LFSCBitVectorProof::printConstant(Expr term, std::ostream& os) {
+void BitVectorProof::printConstant(Expr term, std::ostream& os) {
```

Comment:
[ResolutionBitVectorProof](https://github.com/CVC4/CVC4/blob/609882974d85bb510d9b3256736d0b16a8e63634/src/proof/resolution_bitvector_proof.h#L52) inherits from the current class (`BitVectorProof`), and does not override this function. This suggests that any potential descendent of `ResolutionBitVectorProof` might reuse this function. However, it explicitly prints out `a_bv`, which is unique for LFSC. Should we move it? Or maybe only handle here lfsc proofs (and dismiss the extra layer e.g. in [here](https://github.com/CVC4/CVC4/blob/609882974d85bb510d9b3256736d0b16a8e63634/src/proof/resolution_bitvector_proof.h#L84)) The same holds for other functions here, e.g. `printOperatorNary` that uses `toLFSCKindTerm`.

Reply:
I agree, I would be in favor of removing all LFSC related code from `BitVectorProof` and `ResolutionBitVectorProof`. If additional modifications need to be done first before we can do that, we can also move them in a later PR, though.

Reply:
I agree that it would be good to separate LFSC-printing code from proof-structure-storing code.

However for reasons that will be made clear when a subsequent PR introduces `DRATBitVectorProof`, pulling all the LFSC-printing code to the bottom of the class hierarchy isn't the nicest solution, because most LFSC-printing code isn't resolution or DRAT specific.

My plan is to build a parallel hierarchy of LFSC-printers, which take a `BitVectorProof` as their input.